### ISSUE TITLE
computeCurrentScript() from d8 preamble fails in VS Code extension host (NodeJS) due to tabs instead of spaces for stack trace indenting

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#55684]
I was trying to run some compiled Dart in a VS Code extension (which runs in a NodeJS process) using the [d8 preamable file](https://github.com/dart-lang/sdk/blob/main/sdk/lib/_internal/js_runtime/lib/preambles/d8.js)).

The `computeCurrentScript` function fails like this:

![image](https://github.com/dart-lang/sdk/assets/1078012/22a0a907-7314-4303-ba9b-bf7f47b4713b)

I added some logging of the regex and the stack that was captured:

```js
      var lastMatch = null;
      do {
        console.warn(re.source);
        console.warn(stack);
        var match = re.exec(stack);
        console.warn(match);
        if (match != null) lastMatch = match;
      } while (match != null);
      return lastMatch[1];
```

And the output looked like this:

![image](https://github.com/dart-lang/sdk/assets/1078012/a4fd294d-b006-4730-9ea3-a6b9a9369db6)

Everything appeared good but after some more debugging I found that the stack trace here has **tabs** and not **spaces**!

If I change the start of the regex from `/^ *at` to `/^\s*at` then the code works (or at least it gets further.. but I think the remaining issues are my own).

I cannot figure out why the stack traces have tabs here, but it seems like changing the space to `\s` here would not be breaking so I'm hopeful that it is a change that could be made. I'm happy to open a CL if someone can confirm this is a valid change.

**Comment by sigmundch** [comment]
Wow, that's so bizarre!

I'm not opposed to changing it just to make it more flexible. Here is a CL to do so: https://dart-review.googlesource.com/c/sdk/+/366160

Note: while the d8 preamble is OK to use in node, the [`node_preamble`](https://pub.dev/packages/node_preamble) may be a better fit. That package however has the same logic and currently [uses spaces rather than `\s`](https://github.com/mbullington/node_preamble.dart/blob/47245865175929ec452d8058e563c267b64c3d64/lib/preamble.js#L79), so it would also need to be updated.

**Comment by DanTup** [comment]
> Here is a CL to do so: https://dart-review.googlesource.com/c/sdk/+/366160

Thanks!


> Note: while the d8 preamble is OK to use in node, the [node_preamble](https://pub.dev/packages/node_preamble) may be a better fit.

It actually turned out that the preamble completely broke all the built-in VS Code extensions. It seems like it was manipulating something global that VS Code / other extensions didn't expect that I got lots of exceptions. In the end I ditched the pre-amble and just added the minimum I needed (I think `globalThis.self = globalThis.self ?? {};`) for the code to run.

I don't know what the d8 in that preamble meant, I just saw someone else link to it and tried it. I'm curious what it's intended for / how it differs from the node_preamble package?

If I did need one though, I'd be concerned about depending on a package that just says "unverified uploader" and hasn't been published for 14 months - I assumed what's in the SDK would be more likely to be maintained (but that feeling is probably not based on a whole lot!).
